includes/ExternalRedirectInterceptor.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace NewfoldLabs\WP\Module\Onboarding;
+
+use NewfoldLabs\WP\Module\Onboarding\Data\Data;
+
+/**
+ * Class to intercept redirect calls and filter them.
+ *
+ * The purpose of this class is to prevent any redirects while the user is on the onboarding page.
+ * The only allowed redirect is to the brand plugin page.
+ */
+class ExternalRedirectInterceptor {
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		if ( ! isset( $_GET['page'] ) || \sanitize_text_field( $_GET['page'] ) !== WP_Admin::$slug ) {
+			return;
+		}
+
+		\add_filter( 'wp_redirect', array( $this, 'wp_redirect' ), 10, 1 );
+	}
+
+	/**
+	 * Intercept wp_redirect calls and filter them.
+	 *
+	 * @param string $location The location to redirect to.
+	 */
+	public function wp_redirect( $location ): string {
+		$runtime_data = Data::runtime();
+		$brand_plugin_url  = $runtime_data['currentBrand']['pluginDashboardPage'];
+
+		// Intercept if the redirect is going anywhere other than the brand plugin page.
+		if ( strpos( $location, $brand_plugin_url ) !== 0 ) {
```

Comment:
@wpalani Is the logic here right? If not found, it returns false and since `false !== 0`, it evaluates to true. If found, but not at the beginning, then it evaluates to a number greater than 0.

Reply:
Yes, it's correct; if not found and the value is `false`, we shouldn't redirect. and if found and but the position is not `0` we shouldn't redirect either. Here's a breakdown of the logic:

1. If `$brand_plugin_url` is found at the beginning of `$location`, `strpos($location, $brand_plugin_url)` will return 0, which satisfies !== 0 as `false`. Therefore, it won’t enter the if block, allowing the redirect.
2. If `$brand_plugin_url` is not found at all, `strpos()` will return `false`, and false !== 0 evaluates to true, so the if condition is met, returning `''` and preventing the redirect.
3. If `$brand_plugin_url` is found but not at the beginning, `strpos()` will return a value greater than 0, which also satisfies the !== 0 check, returning `''` and stopping the redirect.

I can def see how this line can be hard to read though (to you point), so I just pushed this small code change to make it more readable:

```php
public function wp_redirect( $location ): string {
  $runtime_data     = Data::runtime();
  $brand_plugin_url = $runtime_data['currentBrand']['pluginDashboardPage'];
  
  $location_is_brand_plugin_url = strpos( $location, $brand_plugin_url );
  // Intercept if the redirect is going anywhere other than the brand plugin page.
  if ( false === $location_is_brand_plugin_url || 0 !== $location_is_brand_plugin_url ) {
	  return '';
  }
  
  // Allow the redirect if it's going to the brand plugin page.
  return $location;
}
```